### PR TITLE
Custom layer

### LINKED ISSUE #12: Allow users to set their own assets base path
At the moment, we expect that the texture file is inside the root of our `assets/`, but int most cases, we have different directories for our assets. So I think it makes sense to allow user to set the default path for the particular assets.

https://github.com/manjav/particular/blob/f522112486b41222175f38986644365083dda266/lib/src/particular_controller.dart#L191-L192

And also, I think we can have the default path like this: `assets/particular/`

### PR DESCRIPTION
**Pull request by manjav**

### CODE REVIEW

File: editor/lib/display/header_view.dart [hunk 1]
```diff
@@ -133,7 +134,9 @@ class _HeaderViewState extends State<HeaderView> {
       var textureName = widget.controller.layers[i].configs.textureFileName;
 
       if (!texures.containsKey(textureName)) {
-        texures[textureName] = widget.controller.layers[i].textureBytes!;
+        texures[textureName] =
+            (widget.controller.layers[i] as ParticularEditorLayer)
```
Comment (imaNNeo) [inline editor/lib/display/header_view.dart:138]: I didn't realize why we need to have an abstraction for `ParticularLayer` and `ParticularEditorLayer`?

Comment (manjav) [reply editor/lib/display/header_view.dart:138]: For holding away unnecessary texture bytes from plugin (memory usage)

Comment (imaNNeo) [reply editor/lib/display/header_view.dart:138]: Ahh okay, you just defined the ParticularEditorLayer inside the `/editor` app. I thought it is inside the plugin.

So it seems good to me.

**Comment by imaNNeo** [comment]
LGTM

**Pull request merged**